### PR TITLE
DWR-652 Add support to handle GroupExamReportRequests

### PR DESCRIPTION
**Pull request by wtritch**
Add support to handle GroupExamReportRequest instances. This re-structures the request handlers a bit.  With the ExamQueryParams object, we don't need individual student id fetchers and report fetchers for handling with a GroupExamReportRequest vs a SchoolGradeExamReportRequest, we just need a transformer that can turn either request type into an ExamQueryParams object.

### CODE REVIEW

File: pdf-report-processor/src/main/java/org/opentestsystem/rdw/reporting/processor/requesthandler/ExamQueryParamsParser.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
+package org.opentestsystem.rdw.reporting.processor.requesthandler;
+
+import org.opentestsystem.rdw.reporting.common.report.AbstractExamReportRequest;
+import org.opentestsystem.rdw.reporting.common.report.ExamQueryParams;
+
+/**
+ * Implementations of this interface are responsible for converting AbstractExamReportRequest
+ * instances into ExamQueryParams instances.
+ */
+public interface ExamQueryParamsParser {
```
Comment (wtritch) [inline pdf-report-processor/src/main/java/org/opentestsystem/rdw/reporting/processor/requesthandler/ExamQueryParamsParser.java]: This is the new per-report-type interface.

File: pdf-report-processor/src/main/java/org/opentestsystem/rdw/reporting/processor/requesthandler/SchoolGradeReportFetcher.java [hunk 2]
```diff
@@ -1,137 +0,0 @@
...
-                .schoolId(request.getSchoolId())
-                .assessmentType(request.getAssessmentType() == null ? null : request.getAssessmentType().getAssessmentType())
-                .gradeId(request.getGradeId())
-                .subject(request.getSubject())
-                .build();
-
-        final Map<Long, StudentCompositeReport> compositeReports = new HashMap<>();
-
-        appendIABReports(compositeReports, permissionScope, studentIds, examQueryParams);
-        appendICAReports(compositeReports, permissionScope, studentIds, examQueryParams);
```
Comment (wtritch) [inline pdf-report-processor/src/main/java/org/opentestsystem/rdw/reporting/processor/requesthandler/SchoolGradeReportFetcher.java]: This functionality was moved into the ReportWriter

File: pdf-report-processor/src/main/java/org/opentestsystem/rdw/reporting/processor/requesthandler/SchoolGradeStudentIdFetcher.java [hunk 3]
```diff
@@ -1,48 +0,0 @@
...
-                .schoolYear(request.getSchoolYear())
-                .schoolId(request.getSchoolId())
-                .assessmentType(request.getAssessmentType() == null ? null : request.getAssessmentType().getAssessmentType())
-                .gradeId(request.getGradeId())
-                .subject(request.getSubject())
-                .build();
-
-        return studentIdRepository.findStudentIdsByExamQueryParams(
-                permissionScope,
-                examQueryParams);
```
Comment (wtritch) [inline pdf-report-processor/src/main/java/org/opentestsystem/rdw/reporting/processor/requesthandler/SchoolGradeStudentIdFetcher.java]: This functionality was moved into the FetchStudentIds tasklet.

File: pdf-report-processor/src/main/java/org/opentestsystem/rdw/reporting/processor/step/ReportStudentWriter.java [hunk 4]
```diff
@@ -56,17 +78,22 @@ public void write(final List<? extends Long> studentIdChunk) throws Exception {
                 .filter(Long.class::isInstance)
                 .map(Long.class::cast)
                 .collect(Collectors.toList());
-        studentReports.putAll(this.reportFetcher.fetchReports(reportRequest, permissionScope, studentIds));
+
+        appendIABReports(studentReports, permissionScope, studentIds, queryParams);
+        appendICAReports(studentReports, permissionScope, studentIds, queryParams);
+        //TODO Append Summitive Reports
```
Comment (malaffoon) [inline pdf-report-processor/src/main/java/org/opentestsystem/rdw/reporting/processor/step/ReportStudentWriter.java]: just to nip it in the bud ... summative

Comment (jtreuting) [inline pdf-report-processor/src/main/java/org/opentestsystem/rdw/reporting/processor/step/ReportStudentWriter.java]: Exactly.  And the plural is summative's

Comment (wtritch) [inline pdf-report-processor/src/main/java/org/opentestsystem/rdw/reporting/processor/step/ReportStudentWriter.java]: fixed

**Pull request merged**